fix: Wrong escape sequences

aragon999 opened the pull request:
Maybe fixes #1004

m4rc1e commented on Lib/gftools/builder/operations/autohintOTF.py:
```diff
@@ -3,4 +3,4 @@
 class AutohintOTF(OperationBase):
     description = "Run otfautohint"
-    rule = "otfautohint $args -o $out $in \|\| otfautohint $args -o $out $in --no-zones-stems"
+    rule = "otfautohint $args -o $out $in \\|\\| otfautohint $args -o $out $in --no-zones-stems"
```
@arrowtype's traceback shows,

```
otfautohint: error: argument -o/--output: /fonts/Familyname/|| is not a valid path to write to.
```

The or operator looks well escaped. The issue is that the or operator is attached to the filename. Perhaps quoting the $out and $in args will work e.g

```
'otfautohint $args -o "$out" "$in" \|\| otfautohint $args -o "$out" "$in" --no-zones-stems'
```

aragon999 replied:
You are right, probably the `$in` and `$out` arguments should be escaped with quotes. I will adjust that. The thing I encountered was actually a Python warning:
![image](https://github.com/user-attachments/assets/10343735-fb79-4557-9aae-cf6bd285ade2)

arrowtype commented:
@m4rc1e Sorry it took so long for me to see your request to try this. I don’t face this issue _all_ the time, and often, it just requires me re-running the build one or two times, and then it works. So, that’s odd, but seems to work.

However, I did test this branch, and it worked the first time, and a second time. The code changes also appear sensible enough to me, FWIW.

aragon999 commented:
I fixed the merge conflict and would be happy if this gets merged, as this also fixes a `SyntaxWarning` since Python 3.12:
```
/usr/lib/python3.13/site-packages/gftools/push/items.py:220: SyntaxWarning: invalid escape sequence '\s'
  return re.sub("\s+", " ", text).strip()
```

As the PR did: https://github.com/googlefonts/gftools/pull/1073

Or should I split this PR up?
